apps/pie-docs/src/components/text-input/overview/overview.md

```diff
@@ -0,0 +1,492 @@
+When used in a form, all text input widths should be aligned.
+
+{% contentLayout %}
+  {% contentItem %}
+    {% contentPageImage {
+      src: "../../../assets/img/components/text-input/size-width-default.svg",
+      width: "256px",
+      alt: "Text input with default width."
+    } %}
+  {% endcontentItem %}
+  {% contentItem %}
+    {% contentPageImage {
+      src: "../../../assets/img/components/text-input/size-width-small.svg",
+      width: "110px",
+      alt: "Text input with small width."
+    } %}
+  {% endcontentItem %}
+{% endcontentLayout %}
+
+---
+
+## Content
+
+### Placeholder
+
+Use clear placeholder text for the text input trigger so that users understand the purpose.
+
+{% contentPageImage {
+    src:"../../../assets/img/components/text-input/content-placeholder.svg",
+    alt: "Text input with placeholder content.",
+    width: "256px"
+} %}
+
+### Input types
+
+{% contentLayout %}
+  {% contentItem %}
+    <h4>Alphanumeric</h4>
+    <p>Both letters, numbers and certain special characters are allowed, for the majority of use-cases this will be used.</p>
+    {% contentPageImage {
+      src: "../../../assets/img/components/text-input/content-inputtypes-alphanumeric.svg",
+      width: "256px",
+      alt: "Text input with alphanumeric content."
+    } %}
+  {% endcontentItem %}
+  {% contentItem %}
+    <h4>Numeric</h4>
+    <p>Only numeric values can be entered, for use-cases such as zip code, phone number, card number.</p>
+    {% contentPageImage {
+      src: "../../../assets/img/components/text-input/content-inputtypes-numeric.svg",
+      width: "135px",
+      alt: "Text input with numeric content."
+    } %}
+  {% endcontentItem %}
+  {% contentItem %}
+    <h4>Password</h4>
+    <p>Both letters and number are replaced with dots that hide password text by default.</p>
+    {% contentPageImage {
+      src: "../../../assets/img/components/text-input/content-inputtypes-password.svg",
+      width: "256px",
+      alt: "Text input with password content."
+    } %}
+  {% endcontentItem %}
+{% endcontentLayout %}
+
+### Formatting
+
+#### Date
+
+A date format can be applied to date entries such as DOB, expiry date, start/end date - a slash is automatically displayed when the user enters a value that fits a date format. The user can only enter numeric values in a date text input.
+We support several types of numeric date formats in engineering. The position of the date, month and year can be interchangeable.
+
+{% contentPageImage {
+    src:"../../../assets/img/components/text-input/content-formatting-date.svg",
+    alt: "Text input with date formatting.",
+    width: "690px"
+} %}
+
+#### Card number
+
+Specifically for card number entry, the card number is displayed in groups of 4 digits. A space appears automatically after the user has entered 4 digits.
+The user can enter a maximum of 16 digits.
```

Comment:
This can be updated later when we come round to building this functionality, but I believe card numbers can be [up to 19 digits](https://www.ansi.org/standards-news/all-news/2016/07/announcing-major-changes-to-the-issuer-identification-number-iin-standard-28#:~:text=length%2C%20ranging%20from-,10%20to%2019%20digits,-.%20This%20change%20is).

Reply:
(We probably won't be building this functionality)

Reply:
I think the way it's written it sounds like something the component will support out-of-the-box. Maybe it needs clarifying. It's _briefly_ mentioned later in the page with "if required is handled by pillar engineers".

Reply:
Yeh, I agree it should likely be clarified. Maybe there should be a section in the code page that refers to Dates and card numbers, saying that they're not a part of the core component and this functionality will be looked at later.

Reply:
a couple of tweaks are going to be added to the documentation but it won't block any of your work and we will just handle it after